### DISCUSSION TITLE
Is it possible to access asset from the other compilation?

### DISCUSSION THREAD

**Discussion by zdm** [webpack/webpack#14465] [Q&A] [answered]
I have two parallel compilations:

1. application itself;
2. web worker;

I need to load worker from the application, and I need worker file name to be hashed.

Is it possible to implement this somehow?

**Comment by alexander-akait** [comment, marked as answer]
You can use https://webpack.js.org/configuration/other-options/#dependencies, and add `dependencies: ['webworker'],` so webpack will run them in properly order

**Comment by zdm** [comment]
I will take a look, thank you very much!

**Comment by zdm** [comment]
`Dependecies` sets the order of the compilation.
But how I can know, that filename exactly was generated (hash) for worker?

**Reply by alexander-akait** [reply]
I recommend to keep filename like `my.worker.js` in `webworker` target, and use `new URL('./dist/worker/my.worker.js', import.meta.url)` in `app` target, so `app` allows your to change worker name using `output.filename`

**Comment by zdm** [comment]
I wanted to use hashed names to avoid problems with http cache.

**Reply by alexander-akait** [reply]
It will be hashed by `app` target

**Comment by zdm** [comment]
hmm, interesting
i will try
thank you

**Comment by zdm** [comment]
So, It works with regular build.
I set up `dependencies` property. At first it builds `worker.js` and put it to the `dist/worker.js.
Then in app code I am using new URL("./dist/worker.js"), it hashing `woker.js`. Need to remove original `worker.js` after compilation somehow.

But this doesn't works when I run it under webpack dev server. Ut complains, that path `./dist/worker.js` is not found.
Webpack dev server stores all assets in the memory, but `new URL()` parser tries to find file in the real filesystem.
Is this a bug or I am doing something wrong?

**Reply by alexander-akait** [reply]
You can use https://github.com/webpack/webpack-dev-middleware#writetodisk, i.e.
```
devServer: { 
  devMiddleware: { writeToDisk: true }
}
```

for `webworker` target

**Comment by zdm** [comment]
Does `new URL()` resolves webpack aliases?
I want to create `output` alias and use it like `new URL("output/js/worker.js", import.meta.url)`.

**Reply by alexander-akait** [reply]
You don't need `aliases` here, but yes you can

**Comment by zdm** [comment]
Worker could be used from the third-party module, which could not knows real `output` directory location. 
Easiest way is to create alias.

**Comment by zdm** [comment]
I am sorry, that disturbing you with the stupid question, but could you, please, help me one more time?
I am unable to find answer in the documentation.

I configured two loaders:
```
// web worker
{
	"test": /\.worker\.js$/,
	"type": "asset/resource",
	"generator": {
		"filename": "[name].[hash][ext][query]",
	},
},

// js
{
	"test": /\.m?jsx?$/,
	"exclude": [],
	"use": [
		{
			"loader": "babel-loader",
			"options": global.WEBPACK.BABEL_OPTIONS,
		},
	],
}
```

I am expecting, that `*.worker.js` files will be processed using first loader only, but it, seems, also processed by second too.
Is it possible to configure some option, that will tell to stop processing file with the other loaders, that can match `test` regexp?

**Reply by alexander-akait** [reply]
exclude: `/\.worker\.js$/`

**Comment by zdm** [comment]
I found `oneOf`.
Thank you for your help.
